apps/omg_child_chain/lib/omg_child_chain/fee_server.ex

```diff
@@ -38,20 +53,14 @@ defmodule OMG.ChildChain.FeeServer do
   def init(args) do
     :ok = ensure_ets_init()
 
-    # This will crash the process by returning {:error, :fee_adapter_not_defined}
-    # if the fee_adapter is not set in the config
-    {:ok, fee_adapter} = get_fee_adapter()
-
     {:ok, state} =
-      args
-      |> Enum.into(%{})
-      |> Map.put(:fee_adapter, fee_adapter)
-      |> Map.put(:expire_fee_timer, nil)
+      __MODULE__
+      |> Kernel.struct(args)
```

Comment:
isn't it an overkill for pipes? shouldn't we rather prefer `struct(__MODULE__ , args) |> ...`?